### PR TITLE
DR | Edit review page focuses on headers

### PR DESCRIPTION
**Pull request by Mottie**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- _(Summarize the changes that have been made to the platform)_
  > - Adding `reviewEditFocusOnHeaders` option to `config/form` to change review page edit focus behavior. Currently, when a page is editing on the review & submit page, the first focusable element is focused.
  > - Also updated SC 5103 behavior to re-focus on the second header after the alert is closed
  > - Fixed area of disagreement focus on header
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits Decision Reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/92801

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
  > Updated and added unit tests
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         |  After |
| ------ | ----- |
| NOD | focus-nod |
| HLR | focus-hlr |
| SC | focus-sc |
| SC prototype | focus-sc-proto |

## What areas of the site does it impact?

- All DR forms
- All forms that set `reviewEditFocusOnHeaders` in the `config/form` to `true`

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by powellkerry — DISMISSED** [review]

File: src/applications/appeals/995/components/PrimaryPhone.jsx [hunk 1]
```diff
@@ -62,7 +62,11 @@ export const PrimaryPhone = ({
   };
 
   const navButtons = onReviewPage ? (
-    <va-button text={content.update} onClick={updatePage} uswds />
+    <va-button
+      text={content.update}
+      onClick={updatePage}
+      label="update primary phone number"
+    />
```
Comment (shaunburdick) [inline src/applications/appeals/995/components/PrimaryPhone.jsx:69]: This change doesn't seem related to the rest of this PR?

File: src/applications/appeals/testing/sc/components/PrimaryPhone.jsx [hunk 2]
```diff
@@ -62,7 +62,11 @@ export const PrimaryPhone = ({
   };
 
   const navButtons = onReviewPage ? (
-    <va-button text={content.update} onClick={updatePage} uswds />
+    <va-button
+      text={content.update}
+      onClick={updatePage}
+      label="update primary phone number page"
+    />
```
Comment (shaunburdick) [inline src/applications/appeals/testing/sc/components/PrimaryPhone.jsx:69]: Same as above comment

File: src/applications/appeals/testing/sc/utils/focus.js [hunk 3]
```diff
@@ -18,7 +19,14 @@ export const focusEvidence = (_index, root) => {
...
+export const focusH3AfterAlert = ({ name, onReviewPage } = {}) => {
+  if (name && onReviewPage) {
+    // name, editing (alert only visible in edit mode), reviewEditFocusOnHeaders
+    const editing = true; // alert only visible in edit mode
+    const reviewEditFocusOnHeaders = true; // from form/config.js setting
+    focusReview(name, editing, reviewEditFocusOnHeaders);
+  } else {
+    scrollTo('topContentElement');
+    focusElement('h3#header');
+  }
```
Comment (shaunburdick) [inline src/applications/appeals/testing/sc/utils/focus.js:31]: I hate that this is copy/pasta'd, I hate it more that the comments and implementation are inconsistent :-(

File: src/applications/appeals/995/components/PrimaryPhone.jsx (hunk 1 above)
Comment (Mottie) [reply src/applications/appeals/995/components/PrimaryPhone.jsx:69]: It fixes an accessibility issue with the "Update page" button on the review and submit page. Without this change, screen readers would only read out the text "Update page" - so it's sorta related in that I found it while testing focus management

File: src/applications/appeals/testing/sc/utils/focus.js (hunk 3 above)
Comment (Mottie) [reply src/applications/appeals/testing/sc/utils/focus.js:31]: DOH, I'll fix it!

Comment (Mottie) [reply src/applications/appeals/testing/sc/utils/focus.js]: I copy/pasta'd the entire SC form into the prototype... once the research is done, I can then perform the arduous process of adding all these changes back into SC behind a feature toggle 🙀 - so keeping the code "nearly" the same may hopefully make it easier to find the changes.

File: src/applications/appeals/995/components/PrimaryPhone.jsx (hunk 1 above)
Comment (Mottie) [reply src/applications/appeals/995/components/PrimaryPhone.jsx:69]: Yeah, all web components are now using USWDS v3, so the prop setting isn't required any more

Comment (shaunburdick) [reply src/applications/appeals/995/components/PrimaryPhone.jsx:69]: Nice, thanks!

**Review by shaunburdick — APPROVED** [review]
LGTM!

**Review by pjhill — APPROVED** [review]

**Review by powellkerry — APPROVED** [review]

**Comment by Mottie** [comment]
Planned to merge today after daily production deploy, but it was [put on hold](https://dsva.slack.com/archives/C5HP4GN3F/p1727802245653169).
